File: Python/Warmup-1/missing_char.py

# this function return a new string where the char at index n has been removed
def missing_char(str, n):
	
	s = ""

	if n >= 0 and n < len(str):
		if n == 0: # n == 0
			s = str[1:]
		elif n == len(str) - 1: # n == last character
			s = str[:-1]
		else: # n == not 0 neither last character
			s = str[:n] + str[n + 1:]
	else:
		print("Wrong index."),

	return s

print(missing_char("kitten", 1))
print(missing_char("kitten", 0))
print(missing_char("kitten", 4))

File: Java/Warmup-1/missingChar.java

public class warmup1 {

	// this function return a new string where the char at index n has been removed
	public String missingChar(String str, int n) {
		
		String s = "";
		
		if (n >= 0 && n <= str.length()) {
			if (n == 0) // n = 0
				s = str.substring(1);
			else if (n == str.length()) // n = last position
				s = str.substring(0, str.length() - 1);
			else { // n = not 0 neither last position
				String s1 = str.substring(0, n);
				String s2 = str.substring(n + 1);
				s = s1 + s2;
			}
		}
		else {
			System.out.print("Wrong index.");
		}
		
		return s;
		
	}
	
	public static void main(String[] args) {
		
		warmup1 o = new warmup1();
		
		String s;
		
		s = o.missingChar("kitten", 1);
		System.out.println(s);
		
		s = o.missingChar("kitten", 0);
		System.out.println(s);
		
		s = o.missingChar("kitten", 4);
		System.out.println(s);
		
	}
	
}
